refinery/core/models.py

```diff
@@ -2234,3 +2244,309 @@ def _get_total_user_logins(self):
         return sum(u.login_count for u in UserProfile.objects.all()) - \
                sum(s.total_user_logins for s in
                    SiteStatistics.objects.exclude(id=self.id))
+
+
+class Event(models.Model):
+    datetime = models.DateTimeField(default=timezone.now)
+    dataset = models.ForeignKey(DataSet, null=True)
+    group = models.ForeignKey(Group, null=True)
+    user = models.ForeignKey(User, null=True)
+    # Null user should not occur in production, but it lets older tests pass.
```

Comment:
A lot of them. :) There are many tests where we create a data set with factory boy, and there's no user set in the middleware. Other possibilities:
- Scott suggested having a default to the anonymous user, but that felt like sweeping it under the rug... In production we should always have a *real* user, and if it's missing, we want to have problems.
- Or, we could set the cuser thread variable in `setUp`s, it would be a lot of changes, but maybe that's the better course?